src/OpenTelemetry.DependencyInjection/Metrics/OpenTelemetryDependencyInjectionMetricsServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="OpenTelemetryDependencyInjectionMetricsServiceCollectionExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Metrics;
```

Comment:
Normally the namespace would match the namespace of the thing being extended. In this case `IServiceCollection`, so `Microsoft.Extensions.DependencyInjection`.

In speaking with @CodeBlanch offline, he said he intentionally chose `OpenTelemetry.Metrics` here because methods like `ConfigureOpenTelemetryMeterProvider` are for more advanced use cases and he wanted to make it a little less likely these extensions would pop up in intellisense for more common use cases.

I think this is good reasoning, however, what if we took it a bit further and named it `OpenTelemetry.Metrics.AdvancedExtensions` (ugh... naming is hard)? Users would have to be pretty intentional that they do in fact want to use them.

Reply:
.NET team has a guidance on naming: https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors#namespace-guidance, according to which they discourage people from using `Microsoft.Extensions.DependencyInjection` as the namespace.

`OpenTelemetry.Metrics.AdvancedExtensions` is good or we could have `OpenTelemetry.AdvancedExtensions.Metrics` 😀

Reply:
Or they could just not use `ConfigureOpenTelemetryMeterProvider` after reading the documentation which would explicitly say it's for library authors?

Reply:
Just to add a little more to the discussion, in addition to changing the namespaces of these methods I also...

* Changed the names. They used to be "ConfigureOpenTelemetryTracing/Metrics" and now they are "ConfigureOpenTelemetryTracerProvider/MeterProvider". I did that to intentionally scare away casual users. Hopefully they don't know what those things are and shy away from these methods in favor of the more friendly `AddOpenTelemetry` in their intellisense.

* Changed the signatures. They used to receive a simple delegate like `Action<TracerProviderBuilder>` now they receive the service provider as well like `Action<IServiceProvider, TracerProviderBuilder>` the goal there was to also scare away casual users looking for something easy 👻 

PS: I am not opposed to moving them again to some sub namespaces, just wanted to point out these changes.

Reply:
> they discourage people from using Microsoft.Extensions.DependencyInjection as the namespace.

Interesting! I thought I've read guidance stating the opposite before. Though maybe this is an exception?

In that case, we probably want to reconsider our use of `Microsoft.Extensions.DependencyInjection` namespace for all the other classes in this PR?

Though what namespace should `StartWithHost` be in? Either `OpenTelemetry.Extensions.DependencyInjection` or `OpenTelemetry.Extensions.Hosting`? If we did `OpenTelemetry.Extensions.Hosting` then I think it would be very hard to find.

Reply:
Hmm... ok so maybe I was mistaken, this PR does not add anything to the Microsoft.Extensions.DependencyInjection namespace

Reply:
> Hmm... ok so maybe I was mistaken, this PR does not add anything to the Microsoft.Extensions.DependencyInjection namespace

I was chatting with @utpilla offline about this. It was not a goal of this work to be complaint with the guidance, but it just kind of happened 🤣 The reason was we now have `AddOpenTelemetry` which returns `OpenTelemetryBuilder`. Extensions like `StartWithHost` target `OpenTelemetryBuilder`. To make it work nicely for users, everything should be in the same namespace. So it was either put everything in `Microsoft.Extensions.DependencyInjection` or everything in `OpenTelemetry`. The later felt more appropriate to me so that's what landed on the PR.

The only thing left in the MEDI namespace should be the hosting extensions which are now marked `Obsolete`.